### PR TITLE
changes for resource cleanup on exit

### PR DESCRIPTION
**Pull request by madanagopalt**

### CODE REVIEW

File: src/platforms/spark/SparkMediaplayer.mjs [hunk 1]
```diff
@@ -1,10 +1,21 @@
 import lng from "wpe-lightning/src/lightning.mjs";
 
-export default class SparkMediaplayer extends lng.Component {
+var appVideoEl = null;
+
+// clear video element resources
+export function clearVideoElementResources() {
```
Comment (npoltorapavlo) [inline src/platforms/spark/SparkMediaplayer.mjs:6]: this duplicates **sparkscene.on('onClose'** handler in SparkMediaplayer.

File: src/platforms/spark/ApplicationTexture.mjs [hunk 2]
```diff
@@ -1,6 +1,22 @@
 import lng from "wpe-lightning/src/lightning.mjs";
 
-export default class ApplicationTexture extends lng.Texture {
+var appTextures = [];
+
+function clearAppTexture(texture) {
```
Comment (npoltorapavlo) [inline src/platforms/spark/ApplicationTexture.mjs:5]: Does the app not destroy on GC, why doing this explicitly?

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

File: src/platforms/spark/SparkMediaplayer.mjs (hunk 1 above)
Comment (madanagopalt) [reply src/platforms/spark/SparkMediaplayer.mjs:6]: yes, updated

Comment (npoltorapavlo) [reply src/platforms/spark/SparkMediaplayer.mjs:6]: I think a better approach is to use **sparkscene.on('onClose'** and avoid static global variables

File: src/platforms/spark/ApplicationTexture.mjs (hunk 2 above)
Comment (madanagopalt) [reply src/platforms/spark/ApplicationTexture.mjs:5]: Seen a reference to sparkwebgl is there inside created context inside initGL/ESMLoader and causing leak. Tried clearing out in initGL.js , asfter which issue is not seen and seen all the App and WebGLTexture objects etc got destroyed from inspector.

**Review by madanagopalt — COMMENTED** [review]
https://github.com/pxscene/pxCore/pull/2233 has changes for cleaning up in onSceneTerminate from context on which lightning app ran

**Comment by madanagopalt** [comment]
Reason for keeping as one onClose handler is because of unable to register handlers called inside constructors of platform, player and apptextures. All returned same hash value and so dnot registered multiple times. So done with above approach of having single handler.

**Pull request closed without merging**